internal/sidekick/rust_release/preflight.go

```diff
@@ -16,56 +16,42 @@ package rustrelease
 
 import (
 	"context"
-	"fmt"
-	"log/slog"
 
-	"github.com/googleapis/librarian/internal/command"
+	"github.com/googleapis/librarian/internal/config"
 	"github.com/googleapis/librarian/internal/git"
-	"github.com/googleapis/librarian/internal/sidekick/config"
+	"github.com/googleapis/librarian/internal/librarian/rust"
+	sidekickconfig "github.com/googleapis/librarian/internal/sidekick/config"
 )
 
-// CargoPreFlight() verifies all the necessary cargo tools are installed.
-func CargoPreFlight(ctx context.Context, config *config.Release) error {
-	if err := command.Run(ctx, cargoExe(config), "--version"); err != nil {
-		return err
-	}
-	tools, ok := config.Tools["cargo"]
-	if !ok {
-		return nil
-	}
-	for _, tool := range tools {
-		slog.Info("installing cargo tool", "name", tool.Name, "version", tool.Version)
-		spec := fmt.Sprintf("%s@%s", tool.Name, tool.Version)
-		if err := command.Run(ctx, cargoExe(config), "install", "--locked", spec); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
 // PreFlight() verifies all the necessary  tools are installed.
-func PreFlight(ctx context.Context, config *config.Release) error {
-	gitExe := gitExe(config)
+func PreFlight(ctx context.Context, sidekickConfig *sidekickconfig.Release) error {
+	gitExe := gitExe(sidekickConfig)
 	if err := git.GitVersion(ctx, gitExe); err != nil {
```

Comment:
do we also run git.GitVersion and git.GitRemoteURL inside internal/librarian/rust before rust.CargoPreFlight? if so, can this be consolidated into one function?